src/components/header/renderer.tsx

```diff
@@ -301,26 +326,30 @@ class Header extends React.Component<Props, State> {
           </PortalWithState>
 
           <PortalWithState closeOnEsc>
-            {({ openPortal, closePortal, isOpen, portal }) => [
-              <span key="0" onClick={openPortal}>
-                {HelpButton}
-              </span>,
-              portal(
-                <div className="modal-background" onClick={closePortal}>
-                  <div className="modal modal-top" onClick={e => e.stopPropagation()}>
-                    <div className="modal-header">
-                      <button className="close-button" onClick={closePortal}>
-                        <X />
-                      </button>
-                    </div>
-                    <div className="modal-body">
-                      <HelpModal />
+            {({ openPortal, closePortal, isOpen, portal }) => {
+              !this._listnerAttached && this.handleHelpModalToggle(event, openPortal, closePortal, isOpen);
```

Comment:
Can you explain what `this._listenerAttached` is ensuring?

Reply:
I am just checking if the listeners are attached , because the render props will attach listener on every render , which causes the page to crash , so.this condition just checks if the the listeners are attached , if no then call the function that attaches listeners otherwise it will just return and the attached listener will do the job .
I will document this later

Reply:
@siddhant1 but `windows.addEventListener` will still attach the listener right?

Reply:
The function `handleHelpModalToggle` is resposible for adding the event listener , on initial render , the events will be attached and the _listnerAttached will be set to true, now the listener wont be reattached on every render because the this._listener will resolve to true , so the expression will not to evaluated , hence no events attaching